### PR TITLE
Add support for JFR thread park event

### PR DESCRIPTION
**Pull request by roberttoyonaga**
### Summary
Add support for the jdk.ThreadPark event.

### CODE REVIEW

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/ThreadParkEvent.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public static void emit(long startTicks, Object obj, long timeout, long until) {
+        if (HasJfrSupport.get()) {
+            emit0(startTicks, obj, timeout, until);
+        }
+    }
+
+    @Uninterruptible(reason = "Accesses a JFR buffer.")
+    private static void emit0(long startTicks, Object obj, long timeout, long until) {
+        if (SubstrateJVM.isRecording() && SubstrateJVM.get().isEnabled(JfrEvent.ThreadPark)) {
+            JfrNativeEventWriterData data = org.graalvm.nativeimage.StackValue.get(JfrNativeEventWriterData.class);
```
Comment (christianhaeubl) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/ThreadParkEvent.java:49]: Please import the class `StackValue`.

**Review by christianhaeubl — COMMENTED** [review]
Thanks, the changes look good. I added a few minor comments.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/ThreadParkEvent.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            JfrNativeEventWriterDataAccess.initializeThreadLocalNativeBuffer(data);
+            JfrNativeEventWriter.beginSmallEvent(data, JfrEvent.ThreadPark);
+            JfrNativeEventWriter.putLong(data, startTicks);
+            JfrNativeEventWriter.putLong(data, JfrTicks.elapsedTicks() - startTicks);
+            JfrNativeEventWriter.putEventThread(data);
+            JfrNativeEventWriter.putLong(data, SubstrateJVM.get().getStackTraceId(JfrEvent.ThreadPark, 0));
+            JfrNativeEventWriter.putClass(data, obj != null ? obj.getClass() : null);
+            JfrNativeEventWriter.putLong(data, timeout);
+            JfrNativeEventWriter.putLong(data, until);
+            JfrNativeEventWriter.putLong(data, obj != null ? Word.objectToUntrackedPointer(obj).rawValue() : 0L);
```
[the remark this answers is not in the document]
Comment (roberttoyonaga) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/ThreadParkEvent.java]: changed!

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/ThreadParkEvent.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    private static void emit0(long startTicks, Object obj, long timeout, long until) {
+        if (SubstrateJVM.isRecording() && SubstrateJVM.get().isEnabled(JfrEvent.ThreadPark)) {
+            JfrNativeEventWriterData data = org.graalvm.nativeimage.StackValue.get(JfrNativeEventWriterData.class);
+            JfrNativeEventWriterDataAccess.initializeThreadLocalNativeBuffer(data);
+            JfrNativeEventWriter.beginSmallEvent(data, JfrEvent.ThreadPark);
+            JfrNativeEventWriter.putLong(data, startTicks);
+            JfrNativeEventWriter.putLong(data, JfrTicks.elapsedTicks() - startTicks);
+            JfrNativeEventWriter.putEventThread(data);
+            JfrNativeEventWriter.putLong(data, SubstrateJVM.get().getStackTraceId(JfrEvent.ThreadPark, 0));
+            JfrNativeEventWriter.putClass(data, obj != null ? obj.getClass() : null);
```
[the remark this answers is not in the document]
Comment (roberttoyonaga) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/ThreadParkEvent.java]: changed!

**Comment by teshull** [comment]
Looks good to me. Will be merged in https://github.com/oracle/graal/pull/5085

**Pull request merged**